cypress/integration/admin/login/login.spec.js

```diff
@@ -17,4 +19,146 @@ context("Login Page", () => {
+                    cy.log('email:' + fullAccessGroupUser.email)
+                    cy.log('psw:' + password)
+                    cy.visit("/");
+                    // Verify the access
+                    cy.findByText(`Hi ${user.firstName} ${user.lastName}!`).should("be.visible");
+                    cy.findByText(/To get started - pick one of the actions below:/i).should(
+                        "be.visible"
+                    );
+
+                    // Should have Page Builder card
+                    cy.findByTestId("admin-welcome-screen-widget-page-builder").within(() => {
+                        cy.findByText("Page Builder").should("be.visible");
+                        cy.findByText(/Build a new Page/i).should("be.visible");
+                    });
+
+                    cy.findByTestId("logged-in-user-menu-avatar").click();
+                    cy.findByText(/Sign out/i).click();
+                    cy.findByText(/sign in/i).should("exist");
+
+                    // activate user password
+                    cy.findByLabelText(/your e-mail/i).type(newUserAccountEmail);
+                    cy.findByLabelText(/your password/i).type("12345678");
+                    cy.findByText(/submit/i).click();
+                    cy.get('div.react-spinner-material').should('be.visible');
+                    cy.get('div.react-spinner-material').should('not.exist');
+                    cy.get('input[type="password"]').click();
+                    cy.get('input[type="password"]').type('12345678');
+                    cy.get('button.webiny-ui-button > span').click(); // SET PASSWORD btn
+
+                    cy.findByTestId("logged-in-user-menu-avatar").click();
+                    cy.findByText(/Sign out/i).click();
+                    cy.findByText(/sign in/i).should("exist");
+
+                    // recover psw
+                    cy.findByText(/forgot password?/i).click();
+                    cy.findByLabelText(/email/i).type(newUserAccountEmail);
+                    cy.findByText(/send me the code?/i).click();
+                    cy.contains('We have sent you a code to reset your password!').should('be.visible');
+
+                    cy.mailosaurGetMessage(serverId, {
+                        sentTo: newUserAccountEmail
+                    }).then(email => {
+                        expect(email.subject).to.equal('Your verification code');
+                        passwordResetCode = email.html.body.match(/(\d+)/)[0];
+
+                        cy.log(passwordResetCode)
+
+                        cy.contains('Password Recovery').should('be.visible');
+                        cy.contains('I got the code!').click();
+                        cy.get('div > input[type="text"]').type(passwordResetCode)
+                        cy.get('div[data-testid="new-password-input"] input').type(newPassword);
+                        cy.get('div[data-testid="retype-password-input"] input').type(newPassword);
```

Comment:
BTW do we really need `div` that is prefixing `[data-testid...` selection?

Instead of `div[data-testid="retype-password-input"]`, can we do just `[data-testid="retype-password-input"]` ? If you ask me, I'd say it's good enough of a selector. If you feel otherwise, feel free to leave it.

Just a nitpick really.

Reply:
Indeed it's not mandatory to have the div itself as well, it's just an approach that I've been using because I consider that it makes the selector a bit more readable (under the hood in terms of performance would be the same as long as we avoid chains like div > div > div[data-testid] and so on). I will update the selector as per your suggestion. Cheers!